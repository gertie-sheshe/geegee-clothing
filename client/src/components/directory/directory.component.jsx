import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import MenuItem from '../menu-item/menu-item.component';

// Selectors
import { selectSections } from '../../redux/directory/directory.selectors';

import { DirectoryMenuContainer } from './directory.styles';

const Directory = ({ sections }) => (
  <DirectoryMenuContainer>
    {sections.map(({ id, ...section }) => (
      <MenuItem key={id} {...section} />
    ))}
  </DirectoryMenuContainer>
);

const mapStateToProps = createStructuredSelector({
  sections: selectSections,
});

export default connect(mapStateToProps)(Directory);
