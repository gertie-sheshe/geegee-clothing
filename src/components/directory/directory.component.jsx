import React from 'react';
import { connect } from 'react-redux';

import MenuItem from '../menu-item/menu-item.component';

import './directory.styles.scss';

const Directory = ({ sections }) => (
  <div className="directory-menu">
    {sections.map(({ id, ...section }) => (
      <MenuItem key={id} {...section} />
    ))}
  </div>
);

const mapStateToProps = state => ({
  sections: state.directory.sections,
});

export default connect(mapStateToProps)(Directory);
