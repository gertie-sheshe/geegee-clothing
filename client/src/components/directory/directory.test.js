import React from 'react';
import { shallow } from 'enzyme';

import { Directory } from './directory.component';

describe('Directory tests', () => {
  it('Should render Directory Component', () => {
    const wrapper = shallow(<Directory sections={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
