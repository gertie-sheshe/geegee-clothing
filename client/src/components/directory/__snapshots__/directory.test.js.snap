// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory tests Should render Directory Component 1`] = `ShallowWrapper {}`;
